pkg/kubernetes/kubernetes.go

```diff
@@ -203,6 +203,14 @@ func (c *ClientsCollection) Get(ctx context.Context, kind, namespace, name strin
 	return client.Get(ctx, name, options, subresources...)
 }
 
+func (c *ClientsCollection) Exists(ctx context.Context, kind, namespace, name string, options metav1.GetOptions, subresources ...string) bool {
+	_, err := c.Get(ctx, kind, namespace, name, options, subresources...)
+	if err != nil { // TODO: check for specific error
+		return false
```

Comment:
It would make sense to return an error here (if it's not 404) such that it's clear why it failed and can be treated as a problem in cluster registry.

Reply:
fixed